app/components/DropZone.js

```diff
@@ -153,7 +153,7 @@ class DropZone extends Component<Props, State> {
                 entries
               </p>
               {dataRowsTo &&
-                (dataRowsTo <= dataRowsFrom || dataRowsTo <= headerRow) && (
+                (dataRowsTo < dataRowsFrom || dataRowsTo < headerRow) && (
```

Comment:
I discovered a legacy bug there. The last part of the condition should be:
`dataRowsFrom < headerRow` as in the other files

Reply:
Actually, no, afaik: check out the error message, and the first part of the validation on line 155 - they both speak about the last row! I think the idea is to make sure the last row is not greater than the header row somehow

Reply:
I still can't see it (Logic is confusing!). I mean, ideally we want to keep the following order:
`headerRow < dataRowsFrom < dataRowsTo`.

If we leave `dataRowsTo < headerRow` we are allowing the following scenario:
`dataRowsFrom < headerRow < dataRowsTo`, and we shouldn't.

Am I still wrong? (Still doubtful but this makes sense to me)

Reply:
Ohhh i get you now. The thing is, we're not showing any error messages for `dataRowsFrom` - the value is validated and it won't let you go on if it's incorrect, you just don't get a message. I'll add one.